src/Integration.Vsix/ErrorList/SonarErrorListDataSource.cs

```diff
@@ -207,11 +212,19 @@ public void Refresh(IEnumerable<string> affectedFilePaths)
                 foreach (var factory in factories)
                 {
                     var snapshot = factory.CurrentSnapshot;
+                    
                     if (snapshot.FilesInSnapshot.Any(snapshotPath => affectedFilePaths.Any(affected => PathHelper.IsMatchingPath(snapshotPath, affected))))
                     {
                         snapshot.IncrementVersion();
                         InternalRefreshErrorList(factory);
                     }
+
+                    var selectedIssue = issueSelectionService.SelectedIssue;
```

Comment:
Disclaimer: I think this is the wrong way to fix this.

The real fix should be deeper and get rid of the following places as well:
* Hiding the issue in the error list when it is not navigable https://github.com/SonarSource/sonarlint-visualstudio/blob/master/src/Integration.Vsix/ErrorList/IssuesSnapshot.cs#L159
* Clearing the selection when an issue is removed https://github.com/SonarSource/sonarlint-visualstudio/blob/branch-taint/src/IssueViz/Selection/SelectedVisualizationValidityMonitor.cs#L64
* This is now the 3rd "patch": clearing the selection when the span is not navigable, which is unique to live issues and does not apply to hotspots/taint.

I think the full fix should be somewhere in the `IssuesSnapshot` domain:
1. Register to NotifyPropertyChanged of the issue visualizations
2. When a span changes:
	2.1 If the span is not navigable: 
		2.1.1 Remove it from the issues list
		2.1.2 Clear the selection service if this issue was selected
	2.2 Increment the snapshot version and notify the error list (which makes the Refresh method on the `IIssueLocationStore` redundant)
3. When the snapshot is disposed -> clear the selection service if an issue in the snapshot was selected

This will allow us to drop all these patches and will centralize the Error List behavior, which is removing the issues when they are not navigable (as opposed to the taint/hotspots where we keep them). We could then also drop the Validity Monitor.

I didn't make this change now A) because we didn't talk about it yet, and B) I spotted this regression ticket and wanted to have a quick fix  proposal for this release.

Reply:
I agree this problem is localised to live issues in the Error List so the fix should be contained within the classes that handle that feature. However, I don't think the snapshot should know anything about selection per se.